modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/auth/SecondaryStoreUserLoginTestCase.java

```diff
@@ -0,0 +1,255 @@
+                keyPositionMap);
+        Assert.assertNotNull(keyValues, "SessionDataKeyConsent key value is null");
+        if (!AUTH_CODE_BODY_ELEMENT.equals(keyValues.get(0).getKey())) {
+            sessionDataKeyConsent = keyValues.get(0).getValue();
+            EntityUtils.consume(response.getEntity());
+            // Authorization
+            testSendApprovalPost();
+        } else {
+            String authorizationCode = keyValues.get(0).getValue();
+            Assert.assertNotNull(authorizationCode, "Authorization code is null.");
+            EntityUtils.consume(response.getEntity());
+        }
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void atEnd() throws Exception {
+
+        stopTomcat();
+        userStoreConfigAdminServiceClient.deleteUserStore(DOMAIN_ID);
+        userMgtClient.deleteUser(PRIMARY_USERNAME);
+        userMgtClient.deleteUser(DOMAIN_ID + "/" + SECONDARY_USERNAME);
+        deleteApplication(playgroundApp);
+        clear();
+    }
+
+    private void sendAuthorizedPost() throws Exception {
+
+        List<NameValuePair> urlParameters = new ArrayList<NameValuePair>();
+        urlParameters.add(new BasicNameValuePair("grantType", OAuth2Constant.OAUTH2_GRANT_TYPE_CODE));
+        urlParameters.add(new BasicNameValuePair("consumerKey", clientID));
+        urlParameters.add(new BasicNameValuePair("callbackurl", PLAYGROUND_APP_CALLBACK_URI));
+        urlParameters.add(new BasicNameValuePair("authorizeEndpoint", OAuth2Constant.APPROVAL_URL));
+        urlParameters.add(new BasicNameValuePair("authorize", OAuth2Constant.AUTHORIZE_PARAM));
+        urlParameters.add(new BasicNameValuePair("scope", ""));
+        HttpResponse response = sendPostRequestWithParameters(client, urlParameters,
+                OAuth2Constant.AUTHORIZED_USER_URL);
+        Assert.assertNotNull(response, "Authorized response is null");
+        Header locationHeader = response.getFirstHeader(OAuth2Constant.HTTP_RESPONSE_HEADER_LOCATION);
+        Assert.assertNotNull(locationHeader, "Authorized response header is null");
+        EntityUtils.consume(response.getEntity());
+        response = sendGetRequest(client, locationHeader.getValue());
+        Assert.assertNotNull(response, "Authorized user response is null.");
+        Map<String, Integer> keyPositionMap = new HashMap<String, Integer>(1);
+        keyPositionMap.put("name=\"sessionDataKey\"", 1);
+        List<DataExtractUtil.KeyValue> keyValues = DataExtractUtil.extractDataFromResponse(response, keyPositionMap);
+        Assert.assertNotNull(keyValues, "sessionDataKey key value is null");
+        sessionDataKey = keyValues.get(0).getValue();
+        Assert.assertNotNull(sessionDataKey, "Session data key is null.");
+        EntityUtils.consume(response.getEntity());
+    }
+
+    private void testSendApprovalPost() throws Exception {
+
+        HttpResponse response = sendApprovalPost(client, sessionDataKeyConsent);
```

Comment:
Rather than making `sessionDataKeyConsent` a object level variable, make it a method level parameter and ask the caller to pass it. It will make the caller logic more clear.

Reply:
Understood. Fixed.